tiling.js

```diff
@@ -2536,7 +2536,15 @@ function insertWindow(metaWindow, {existing}) {
             connectSizeChanged();
             Scratch.makeScratch(metaWindow);
             if (scratchIsFocused) {
-                Main.activateWindow(metaWindow);
+                /**
+                 * fix for potential mutter change/regression (currently seeing in
+                 * GNOME 43.1.  See https://github.com/paperwm/PaperWM/issues/448).
+                 * Now using "transitions-completed" signal before
+                 * activating new window on scratch layer.
+                 */
+                signals.connectOneShot(actor,'transitions-completed', () => {
+                    Main.activateWindow(metaWindow);
+                });
```

Comment:
Is this the kind of bug that might pop up again in other places? Does it make sense to pull this out into a helper function, or is it not really re-usable? My main concern is how the accumulation of workarounds like this might make this function hard to understand or change. In other words, where is the right place for this complexity to live?

I have no knowledge of gnome-shell internals and very little of PaperWM, so these are actual questions, not opinions-disguised-as-questions. If this is the right place, that's good; I only want to make sure it was considered.

Reply:
This is a really good question - as far as I can tell, the error/issue is only observed with floating windows and not those managed while tiled in PaperWM.  Given this, and that it's pretty isolated to such a use-case , a helper function would also likely be pretty isolated.  

Certainly I agree though (and it woudn't hurt) - I could implement a helper function for activating windows (e.g. `Main.activateWindowAfterTransitions(metaWindow)` or something).  At least I could then add some function comments referencing the issue in one spot etc.).

Reply:
Thanks @smichel17 this one should be good to go now.  Created the function that can be reused to activate window only after transitions are complete.